kong/consumer_group_test.go

```diff
@@ -242,3 +242,112 @@ func compareConsumerGroups(expected, actual []*ConsumerGroup) bool {
 
 	return (compareSlices(expectedNames, actualNames))
 }
+
+func TestConsumerGroupGetEndpointPreGW39(t *testing.T) {
+	RunWhenEnterprise(t, ">=2.7.0 <3.9.0", RequiredFeatures{})
+	assert := assert.New(t)
+	require := require.New(t)
+
+	client, err := NewTestClient(nil, nil)
+	require.NoError(err)
+	assert.NotNil(client)
+
+	// create consumer
+	consumer := &Consumer{
+		Username: String("foo"),
+	}
+	createdConsumer, err := client.Consumers.Create(defaultCtx, consumer)
+	require.NoError(err)
+	assert.NotNil(createdConsumer)
+	t.Cleanup(func() {
+		if createdConsumer != nil {
+			assert.NoError(client.Consumers.Delete(defaultCtx, createdConsumer.ID))
+		}
+	})
```

Comment:
Nit: potential panics when `createdConsumer` is `nil` can still happen but that's not something that we have to worry about that much.